spanner/api/Spanner.Samples/RetryRobot.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright 2021 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START spanner_retry_strategy]
+
+using System;
+using System.Threading;
+
+public class RetryRobot
```

Comment:
@skuruppu Do we need this Sample?

Reply:
@skuruppu ping on this one, thanks!

Reply:
I'm afraid that there's a whole section on the [Getting Started guide](https://cloud.google.com/spanner/docs/getting-started/csharp#performing_retries_on_read-write_transactions) which specifically talk about retries in C#. So we'll need to keep this.

Reply:
Can we at least add the Spanner specific exceptions that are retryable. As far as I can see this is just a generic retry code.

Reply:
Done!

Reply:
@skuruppu Looking at the docs that you linked above, I really think we should be showing a retriable transaction example instead of this retry robot. It might be that the retry robot was implemented before we had the retriable transaction, but it seems outdated now. WDYT?

Reply:
Hmmm this is weird in the sense that such a retry example doesn't exist for other languages. It's possible it's because they all retry aborted transactions internally.

I don't think we need this example at all. Instead, what I'll do is to modify one of the other samples on that page to use a retriable transaction and modify the text to say something like "BTW, this is how you can retry Aborted transactions".

So @laljikanjareeya please delete this file from this PR.

Reply:
Removed!